clients/rust/src/lib.rs

```diff
@@ -402,6 +330,199 @@ impl From<CommitConfig> for sysadminctl::CommitConfig {
     }
 }
 
+/// Constructs a command that takes no values.
+/// The response type is also a param
+macro_rules! no_arg_command {
+    ($name:ident, $buf_type:ty, $return_type:ty) => {
+
+        #[derive(Debug, Clone, PartialEq, Deserialize, Serialize)]
+        pub struct $name {}
+
+        impl $name {
+            pub fn new() -> $name {
+                $name{}
+            }
+
+            fn into_buf(self) -> $buf_type {
+                let buf: $buf_type = ::std::default::Default::default();
+                buf
+            }
+
+            pub fn send_command(self, client: &mut SysadminClient)
+            -> std::result::Result<$return_type, error_chain_generated_errors::Error> {
```

Comment:
I think you'll want `$create::error_chain...` for this to work everywhere within the crate.

Reply:
You'll also want to do the same to `SysadminClient`

Reply:
I don't follow. Why wouldn't it work? What is `$create` ?

Reply:
@kphelps can you help me understand this?

Reply:
Sorry, I made a typo. Its `$crate`. If you use this macro in another module it won't always work because of how macros resolve modules. $crate is a way to ensure it works everywhere. https://doc.rust-lang.org/1.7.0/book/macros.html#the-variable-crate

Reply:
Ahh. That's good to know.